x-pack/plugins/enterprise_search/public/applications/app_search/components/engine/engine_router.tsx

```diff
@@ -68,12 +68,19 @@ export const EngineRouter: React.FC = () => {
 
   const { engineName: engineNameFromUrl } = useParams() as { engineName: string };
   const { engineName, dataLoading, engineNotFound } = useValues(EngineLogic);
-  const { setEngineName, initializeEngine, clearEngine } = useActions(EngineLogic);
+  const { setEngineName, initializeEngine, pollEmptyEngine, stopPolling, clearEngine } = useActions(
+    EngineLogic
+  );
 
   useEffect(() => {
     setEngineName(engineNameFromUrl);
     initializeEngine();
-    return clearEngine;
+    pollEmptyEngine();
+
+    return () => {
+      stopPolling();
+      clearEngine();
+    };
```

Comment:
Here's a screenshot of me QAing that polling stops/starts when switching engines as expected:

![polling](https://user-images.githubusercontent.com/549407/123028704-b168fb80-d394-11eb-9af5-35b0e9380a9f.gif)

(at the end I also test it stops totally by navigating to the Engines page)

Reply:
Doh, the gif cut off so here's a 2nd screencap of me QAing that polling completely stops when navigating away from an empty engine:

![pollingstop](https://user-images.githubusercontent.com/549407/123028991-39e79c00-d395-11eb-9fed-0a94e7a009ce.gif)